plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveViewReaderFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.MetadataProvider;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.type.TypeManager;
+
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+public interface HiveViewReaderFactory
+{
+    ViewReader createViewReader(
+            SemiTransactionalHiveMetastore metastore,
+            ConnectorSession session,
+            Table table,
+            TypeManager typeManager,
+            BiFunction<ConnectorSession, SchemaTableName, Optional<CatalogSchemaTableName>> tableRedirectionResolver,
+            MetadataProvider metadataProvider,
+            boolean runHiveViewRunAsInvoker,
+            HiveTimestampPrecision hiveViewsTimestampPrecision);
```

Comment:
TypeManager - can be injected to the implementation right ? Same for MetadataProvider, runHiveViewRunAsInvoker, and HiveTimestampPrecision.